pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
+										ReadOnly:  true,
+									},
+								},
+								Resources: newBasicRack().Resources,
+							},
+						},
+						DNSPolicy:          "ClusterFirstWithHostNet",
+						ServiceAccountName: "basic-member",
+						Affinity:           &corev1.Affinity{},
+					},
+				},
+				VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   "data",
+							Labels: newBasicStatefulSetLabels(),
+						},
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+							StorageClassName: newBasicRack().Storage.StorageClassName,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceStorage: resource.MustParse(newBasicRack().Storage.Capacity),
+								},
+							},
+						},
+					},
+				},
+				ServiceName:         "basic-client",
+				PodManagementPolicy: appsv1.OrderedReadyPodManagement,
+				UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilImagePullSecrets := func() []corev1.LocalObjectReference {
+		return []corev1.LocalObjectReference{
+			{
+				Name: "basic-secrets",
+			},
+		}
+	}
+
+	newNonNilTolerations := func() []corev1.Toleration {
+		return []corev1.Toleration{
+			{
+				Key:      "key",
+				Operator: corev1.TolerationOpEqual,
+				Value:    "value",
+				Effect:   corev1.TaintEffectNoSchedule,
+			},
+		}
+	}
+
+	newNonNilNodeAffinity := func() *corev1.NodeAffinity {
+		return &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "key",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"value"},
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilPodAffinity := func() *corev1.PodAffinity {
+		return &corev1.PodAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilPodAntiAffinity := func() *corev1.PodAntiAffinity {
+		return &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+				{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"key": "value",
+						},
+					},
+				},
+			},
+		}
+	}
+
+	tt := []struct {
+		name                string
+		rack                scyllav1.RackSpec
+		scyllaCluster       *scyllav1.ScyllaCluster
+		existingStatefulSet *appsv1.StatefulSet
+		sidecarImage        string
```

Comment:
sidecarImage is always set to `""`, so you could just give the function a value (ideally something image like)

Reply:
fixed